base/lock.jl

```diff
@@ -9,6 +18,48 @@ type ReentrantLock
     ReentrantLock() = new(nothing, Condition(), 0)
 end
 
+"""
+    islocked(the_lock) -> Status (Boolean)
+
+Check whether the lock is held by any task/thread.
+This should not be used for syncronization (see instead `trylock`).
+"""
+function islocked(rl::ReentrantLock)
+    return rl.reentrancy_cnt != 0
+end
+
+"""
+    trylock(the_lock) -> Success (Boolean)
+
+Acquires the lock if it is available,
+returning `true` if successful.
+If the lock is already locked by a different task/thread,
+return `false`.
+
+Each successful "trylock" must be matched by an "unlock"
+"""
+function trylock(rl::ReentrantLock)
+    t = current_task()
+    if rl.reentrancy_cnt == 0
+        rl.locked_by = t
```

Comment:
and when you say "known feature," clearly that's not universally true, as that's a subtle few lines of code and very easy to miss without knowing it exists. if you're using Nullable as a data container for quick scripts that no one else will read, sure optimize for conciseness. but when Nullable is an important part of the API design in a core library that many people will need to read, just spell it out - follows very much with the "talking about types" philosophy